### PR TITLE
Switch from an initializer to manual `extendResolver` usage

### LINKED ISSUE #185: Incompatibility with `ember-resolver` v13.x
## Description: 
After upgrading `ember-resolver` to version 13.x, the application encounters an error that prevents it from running correctly when using ember-can. The error message is as follows:
```js
ember_resolver__WEBPACK_IMPORTED_MODULE_0__.default.reopen is not a function
    at webpack://__ember_auto_import__/./node_modules/ember-can/dist/initializers/setup-ember-can.js:7:56
```

## Steps to Reproduce:

 * Upgrade `ember-resolver` to version 13.x in an Ember.js application.
 * Include ember-can in the application.
 * Run the application and observe the error.

## Expected Behavior: 
The application should run without any errors when using ember-can with the updated ember-resolver package.

## Actual Behavior: 
The application fails to run due to the above-mentioned error.

## Environment:

Ember.js version: 5.12
`ember-resolver` version: 13.0.1
`ember-can` version: 6.0.0
Node.js version: 20.x

### PR DESCRIPTION
**Pull request by mkszepp**
`ember-resolver` v13 is now a native class instead of a `EmberObject`.

With `EmberObject` there was possible to `reopen` the class for making changes to `pluralizedTypes`.
This isn't anymore working with native classes. For this reason we need to provide users a short instructions like they can configure `ember-can` inside there app. 

fix #185, close #184
implement logic from comment https://github.com/minutebase/ember-can/pull/184#issuecomment-2357658481

The app.js / .ts looks at end like:
```js
import Application from '@ember/application';
import Resolver from 'ember-resolver';
import loadInitializers from 'ember-load-initializers';
import config from 'my-app/config/environment';
import { extendResolver } from 'ember-can';

export default class App extends Application {
  modulePrefix = config.modulePrefix;
  podModulePrefix = config.podModulePrefix;
  Resolver = extendResolver(Resolver);
}

loadInitializers(App, config.modulePrefix);
```

Why `import { extendResolver } from 'ember-can';` instead of `import Resolver from 'ember-can';`?
The advantage is, that when an app has already extended the `Resolver` he can pass his extended resolver and the extending order is not relevant.

PR changes:
- Add function for app
- Add tests for `ember-resolver` < 12 with current latest ember-source LTS
- Add instructions for users in install documentation

### CODE REVIEW

**Review by wozny1989 — APPROVED** [review]
Nice @mkszepp, looks very nice! 👍

File: ember-can/src/-private/resolver.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type Resolver from 'ember-resolver';
+
+export default function extendResolver(
+  resolver: typeof Resolver,
+): typeof Resolver {
+  return class EmberCanResolver extends resolver {
```
Comment (RobbieTheWagner) [inline ember-can/src/-private/resolver.ts:6]: Would this part not also work with earlier resolver versions? I think I would prefer a breaking release and doing this for everyone over conditionally using `reopen`.

Comment (mkszepp) [reply ember-can/src/-private/resolver.ts:6]: @RobbieTheWagner yes it works also with old versions... okay i will change it so that is braking

Comment (mkszepp) [reply ember-can/src/-private/resolver.ts:6]: @RobbieTheWagner done, changes are now braking

**Review by RobbieTheWagner — APPROVED** [review]
This is awesome. Great work @mkszepp, thank you!

**Pull request merged**